### ISSUE TITLE
In `resolve_qualified_import_name`, prefer direct vs. attribute names

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/ruff#10479]
If both `import logging` and `from logging import error` are in scope, and the user asks for the symbol `logging.error`, I _think_ it makes sense to prefer `"error"` over `"logging.error"`.

**Comment by dhruvmanila** [comment]
Yeah, I agree. If the symbol is already imported then we should use it directly instead.
